### DISCUSSION TITLE
How do you set default value to required link

### DISCUSSION THREAD

**Discussion by dodyg** [geldata/gel#4950] [Q&A]
![newbie](https://user-images.githubusercontent.com/1761249/215273034-ebae69f0-4a58-4308-b360-dc7192bac9af.png)

I am following the documentation at https://www.edgedb.com/docs/intro/schema and trying the code out. 

So before I added the `required link stats`, I inserted a record already to the database. After this, my attempt to run a migration after adding the required link fails because it asks for an expression to fill the existing data. What kind of expression shall I supply? Or even better, is there any default value for link object?

**Comment by raddevon** [comment]
The expression will depend on what you want to set the value at for your existing objects. You could use the expression to insert a new `MovieStats` object. If you have only a single movie, that would work. Just insert the `MovieStats` object for the `Movie` you've already inserted. If you have more than one, you could always insert a "dummy" `MovieStats` record and then replace it individually for each `Movie` after the migration completes.

**Reply by dodyg** [reply]
How do you actually insert a "dummy" one in the definition. I tried this one for example and it fails.

```
module default {
  type Person {
    required property name -> str;
  }

  type Movie {
    required property title -> str {
        constraint exclusive;
        constraint min_len_value(3);
    };
    property uppercase_title := str_upper(.title);
    multi link actors -> Person;
    property release_year -> int64;
    link director -> Person;
    required link stats -> MovieStats {
        constraint exclusive;
        default:= (INSERT MovieStats { budget := 0, box_office := 0 });
    }
  }

  type MovieStats {
    required property budget -> int64;
    required property box_office -> int64;
  }
}

```

**Reply by raddevon** [reply]
If you move your `MovieStats` type above `Movie`, this ☝️  will work.

```
module default {
  type Person {
    required property name -> str;
  }

  type MovieStats {
    required property budget -> int64;
    required property box_office -> int64;
  }

  type Movie {
    required property title -> str {
        constraint exclusive;
        constraint min_len_value(3);
    };
    property uppercase_title := str_upper(.title);
    multi link actors -> Person;
    property release_year -> int64;
    link director -> Person;
    required link stats -> MovieStats {
        constraint exclusive;
        default:= (INSERT MovieStats { budget := 0, box_office := 0 });
    }
  }
}
```
